packages/core/contracts/financial-templates/perpetual-multiparty/PerpetualCreator.sol

```diff
@@ -65,7 +70,25 @@ contract PerpetualCreator is ContractCreator, Testable, Lockable {
      * @return address of the deployed contract.
      */
     function createPerpetual(Params memory params) public nonReentrant() returns (address) {
-        address derivative = PerpetualLib.deploy(_convertParams(params));
+        // Create a new synthetic token using the params.
+        require(bytes(params.syntheticName).length != 0, "Missing synthetic name");
+        require(bytes(params.syntheticSymbol).length != 0, "Missing synthetic symbol");
+        TokenFactoryExclusiveMinter tf = TokenFactoryExclusiveMinter(tokenFactoryAddress);
+        ExpandedIERC20ExclusiveMinter tokenCurrency = tf.createToken(
+            params.syntheticName,
+            params.syntheticSymbol,
+            IERC20Standard(params.collateralAddress).decimals()
+        );
+        require(
```

Comment:
This check might be redundant, but this contract byetcode is very small so I wasn't very pushed to remove it.